### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by svenwltr**
I heard there is a thing called "testing" and I wanted to try it. I had to refactor the api package a bit to make it testable.

Also I fixed some issues:

* kubernetes-deployment errors now, if no files was rendered
* improved to path comparison (actually only relevant for tests, but avoids issues if paths in the GitHub client changes)

### CODE REVIEW

**Comment by svenwltr** [comment]
@rebuy-de/prp-kubernetes-deployment Please review.

File: pkg/api/test-fixtures/repos/foobar/master/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+hello!
```
Comment (stephanlindauer) [inline pkg/api/test-fixtures/repos/foobar/master/README.md]: :wave:

File: pkg/api/util_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+var (
+	UpdateGolden = flag.Bool("update-golden", false,
```
Comment (stephanlindauer) [inline pkg/api/util_test.go]: golden?

Comment (svenwltr) [inline pkg/api/util_test.go]: There is an interesting talk from Mitchell Hashimoto about this: https://speakerdeck.com/mitchellh/advanced-testing-with-go?slide=19

Comment (stephanlindauer) [inline pkg/api/util_test.go]: :+1: coolo. i'll check it out later.

**Pull request merged**